### PR TITLE
Feature/create user

### PR DESCRIPTION
**Pull request by santiagoRD**
## Summary

create user service integrated with useRequest, added loader and error message on post failure

## Screenshots

integration

## Trello Card

https://trello.com/c/NXhgOMTJ/91-sign-up-ii-integration

### CODE REVIEW

File: src/config/api.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { create } from 'apisauce';
+import { CamelcaseSerializer, SnakecaseSerializer } from 'cerealizr';
 
-const baseURL = 'http://wolox.com';
+const baseURL = 'https://books-training-rails.herokuapp.com/api/v1';
```
Comment (CrisVega08) [inline src/config/api.js]: Use environment variable

File: src/types/types.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 export interface User {
-  name: string;
+  firstName: string;
   lastName: string;
   email: string;
   password: string;
   confirmPassword: string;
+  locale?: string;
```
Comment (CrisVega08) [inline src/types/types.ts]: Is optional ?

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/services/UserService.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import api from 'config/api';
+import { User } from 'types/types';
+
+const createUser = (data: User) => api.post('/users', data);
+
+const UserService = {
```
Comment (manugo-dev) [inline src/services/UserService.ts:6]: I really like this way to export the service!

File: .env [hunk 4]
```diff
@@ -0,0 +1 @@
+REACT_APP_BASE_URL = 'https://books-training-rails.herokuapp.com/api/v1'
```
Comment (CrisVega08) [inline .env:1]: This file must be added to .gitignore

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/screens/SignUp/index.tsx [hunk 5]
```diff
@@ -6,6 +6,9 @@ import InputText from 'components/Input';
 import { email } from 'utils/inputValidations';
 import { I18N_CONFIG, SIGNUP_BUTTONS, SIGNUP_FIELDS, ERROR_MESSAGES } from 'constants/index';
 import { User } from 'types/types';
+import { useLazyRequest } from 'hooks/useRequest';
+import UserService from 'services/UserService';
+import Loading from 'components/Spinner/components/loading';
 
 import wLogo from '../../assets/wLogo.png';
```
Comment (manugo-dev) [inline src/screens/SignUp/index.tsx]: I believe you can use:

`import wLogo from 'assets/wLogo.png';`

**Review by manugo-dev — APPROVED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**